### PR TITLE
docs: Document ConfigMap option changes in v1.4

### PR DESCRIPTION
**Pull request by joestringer**
Fixes: #6590

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6835)

### CODE REVIEW

**Comment by joestringer** [comment]
test-docs-please

File: Documentation/install/upgrade.rst [hunk 1]
```diff
@@ -277,6 +277,52 @@ Changes that may require action
...
+    of up-front memory allocation for entries in BPF maps. If this value is
+    modified, then during the next Cilium startup the restore of existing
+    endpoints and tracking of ongoing connections may be disrupted.This may
+    lead to policy drops or a change in loadbalancing decisions for a
+    connection for some time. Endpoints may need to be recreated to restore
+    connectivity. If this option is set to "false" during an upgrade from 1.3
+    or earlier to 1.4 or later, then it may cause one-time disruptions during
+    the upgrade.
+
+  * ``flannel-master-device``: When running Cilium with policy enforcement
```
Comment (joestringer) [inline Documentation/install/upgrade.rst]: Should we wrap the flannel options in a `Preview mode:` indent?

File: Documentation/install/upgrade.rst [hunk 2]
```diff
@@ -277,6 +277,52 @@ Changes that may require action
...
+    enabled on top of a CNI plugin, the BPF programs will be installed on the
+    network interface specified in this option and on each network interface
+    belonging to a pod.
+
+  * ``flannel-uninstall-on-exit``: If ``flannel-master-device`` is specified,
+    this determines whether Cilium should remove BPF programs from the master
+    device and interfaces belonging to pods when the Cilium `DaemonSet` is
+    deleted. If true, Cilium will remove programs from the pods.
+
+  * ``flannel-manage-existing-containers``: On startup, installs a BPF program
```
Comment (joestringer) [inline Documentation/install/upgrade.rst]: Are all of these options intended to stay as `flannel-xxx`, given that there's already talk of integrating with other CNI? Should we rename these first?

**Review by ianvernon — APPROVED** [review]

**Comment by ianvernon** [comment]
Doc test failure is unrelated:
```
01:45:07 Please fix the following spelling mistakes:
01:45:07 * Documentation/cmdref/cilium.md:28: (fqdn) 
01:45:07 * Documentation/cmdref/cilium_fqdn.md:3: (fqdn) 
01:45:07 * Documentation/cmdref/cilium_fqdn.md:5: (fqdn) 
01:45:07 * Documentation/cmdref/cilium_fqdn.md:9: (fqdn) 
01:45:07 * Documentation/cmdref/cilium_fqdn.md:32: (fqdn) 
01:45:07 * Documentation/cmdref/cilium_fqdn_cache.md:3: (fqdn) 
01:45:07 * Documentation/cmdref/cilium_fqdn_cache.md:5: (fqdn) 
01:45:07 * Documentation/cmdref/cilium_fqdn_cache.md:9: (fqdn) 
01:45:07 * Documentation/cmdref/cilium_fqdn_cache.md:31: (fqdn) 
01:45:07 * Documentation/cmdref/cilium_fqdn_cache.md:32: (fqdn) 
01:45:07 * Documentation/cmdref/cilium_fqdn_cache_clean.md:3: (fqdn) 
01:45:07 * Documentation/cmdref/cilium_fqdn_cache_clean.md:5: (fqdn) 
01:45:07 * Documentation/cmdref/cilium_fqdn_cache_clean.md:9: (fqdn) 
01:45:07 * Documentation/cmdref/cilium_fqdn_cache_clean.md:33: (fqdn) 
01:45:07 If the words are not misspelled, add them to Documentation/spelling_wordlist.txt.
```

File: Documentation/install/upgrade.rst (hunk 2 above)
Comment (aanm) [reply Documentation/install/upgrade.rst]: No, they are intended to be kept with that name.

File: Documentation/install/upgrade.rst [hunk 3]
```diff
@@ -277,6 +277,52 @@ Changes that may require action
...
+    to allow for policy enforcement on pods that are currently managed by
+    Flannel. This also requires the Cilium `DaemonSet` to be running with
+    ``hostPID: true``, which is not enabled by default.
+
+Deprecated ConfigMap Options
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (aanm) [inline Documentation/install/upgrade.rst:318]: add the `disable-ipv4`

**Review by aanm — DISMISSED** [review]

**Comment by joestringer** [comment]
test-docs-please

**Comment by joestringer** [comment]
test-docs-please

**Pull request merged**